packages/vc-export/src/vc-js/suites/KiltAttestationProofV1.ts

```diff
@@ -162,10 +162,10 @@ export class KiltAttestationV1Suite extends LinkedDataProof {
   /**
    * Initializes a proof for a [[KiltCredentialV1]] type document.
    *
-   * _! This is not a complete proof yet !_.
+   * _! TO BE VERIFIABLE WITH THIS PROOF, ADJUSTMENTS HAVE TO BE MADE TO THE DOCUMENT !_.
    *
-   * After adding the proof stub to the credential, the resulting document must be processed
-   * by the `finalizeProof` method to make necessary adjustments to the document itself.
+   * To do so, the document, with the proof added, must be processed
```

Comment:
Would it make sense to expose one additional method that returns an object that the `createProof` proof then takes? So instead of exposing a `createProof` and saying that the `finalizeProof` must be called, having a method that returns a `NonFinalizedProof`, which is then passed to the `createProof` to finalize the object? So basically the first method does what the `createProof` does now, and `createProof` would do what `finalizeProof` does now.

In short, relying on the static type system TS provides to make sure that a user won't be able to use a half-created proof.

Reply:
If you look at the implementation in `jsonld-signatures`, this is called only with the following arguments:
```js
    const proof = await suite.createProof({
      document: input, purpose, documentLoader
    });
```
-> there is no good way to pass additional input besides the document to be signed, which also cannot be altered in the signing process. I'm saying no good way bc you could probably craft a `purpose` that contains the non-finalised proof, but then again, where is the point if the result is a document that still can't be verified with that proof?

The changes I proposed make the suite class stateless. If you want to swap the order of things, I could offer that we keep it stateful and require users to call some method (maybe `anchorCredential`) before using `createProof` or functions that rely on it (jsigs `sign` & vc-js `issue`). In that case, the story would be 'you first anchor the credential on the blockchain, then you create a proof over it being anchored`. Note that in this case, `anchorCredential` would be the step at which an attestation is made, and no longer `createProof`. The class instance would then keep track of data required to create the proof (block of inclusion, nonces, etc) and look them up on creating a proof.

Reply:
I think it would be best, if not possible to have compilation-time errors, to have runtime errors before the user sends anything to the verifier. Hence, we should aim to provide a guarantee that "if the proof has been generated and sent following the logic in this SDK, it won't fail to pass the verifier's checks (except revocation etc etc)".
So it's probably fine to have one more step in between, rather than entirely relying on documentation, as is the case now, AFAICT.

Reply:
a pretty easy fix for producing compile-time errors if you try to send off a credential that has not yet been run through `finalizeProof` would be creating proofs where the type field is something else (e.g. `NonFinalizedKiltAttestationProof`), which is then changed once processed. Personally, I don't really like that solution though... It seems like a more straightforward interface to require preprocessing/anchoring first, and then create the proof after the fact. If you get past this point you are sure to have a valid credential.